firefox-ios/Storage/Rust/RustLogins.swift

```diff
@@ -1144,5 +913,19 @@ public class RustLogins: LoginsProtocol {
                 completion(.failure(LoginEncryptionKeyError.illegalState as NSError))
             }
         }
-    }
+
+        // MARK: - KeyManager
+        // This method will be called internally by rust when it needs to encrypt/decrypt logins.
+        // NOTE: Since this is called internally by rust and each CRUD operation will acquire a mutex lock on the db
+        // before doing anything we must make sure that getStoredKey is called before doing any CRUD operation in swift.
+        public func getKey() throws -> Data {
```

Comment:
Note: should put an example somewhere or show how this works for reference

Reply:
Good call.